overlord/fdestate/backend/reseal.go

```diff
@@ -100,93 +104,87 @@ func getUniqueModels(bootChains []boot.BootChain) []secboot.ModelForSealing {
 	return models
 }
 
-func resealKeyForBootChainsFDEHook(method device.SealingMethod, rootdir string, inputs resealInputs, opts resealOptions) error {
-	params := inputs.bootChains
+func doReseal(manager FDEStateManager, method device.SealingMethod, rootdir string) error {
+	hasRunParameters, _, runModels, runPCRProfile, err := manager.Get("run+recover", "all")
+	if err != nil {
+		return err
+	}
+
+	hasRecoverParameters, _, recoveryModels, recoveryPCRProfile, err := manager.Get("recover", "system-save")
```

Comment:
separate thought on "recover", "system-save" bare strings parameters. I recall we discussed having types for them, but what if we passed those as a tuple in a struct? (not for this PR ofc)

Reply:
That would work.